### PR TITLE
Empty collision checker template for usage with tesseract and bullet

### PR DESCRIPTION
**Pull request by j-petit**
### Description
This PR is concerned with GSoC #1427 and supersedes #1484 as the PR started to get too big to be easily reviewed. In this PR no collision checking functionality is included as it is an empty template. The follow up PR #1500 adds a basic Bullet + Tesseract collision checker.

### CODE REVIEW

**Comment by BryceStevenWilley** [comment]
Looks like catkin lint is failing because you have to manually add this new folder to the `moveit_core` [CMakeLists](https://github.com/ros-planning/moveit/blob/898cdc8d8df5689319cd462c2c674954c6134604/moveit_core/CMakeLists.txt#L197).

File: moveit_core/collision_detection_bullet/README.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+### Using Bullet as a collision checker [experimental]
+To use Bullet as a collision checker you can:
+a) create a new planning scene and manually switch to Bullet using the planning_scene.setActiveCollisionDetector()
+b) manually switch to Bullet using the `CollisionPluginLoader`
+c) manually instantiate a `CollisionRobotBt` and `CollisionWorldBt`
+d) use rosparam to set the collision checker (preferred but not tested yet) when starting a new `move_group`
```
Comment (davetcoleman) [inline moveit_core/collision_detection_bullet/README.md]: great docs!!

File: moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/*********************************************************************
+ * Software License Agreement (BSD License)
+ *
+ *  Copyright (c) 2019, Open Source Robotics Foundation, Inc.
```
Comment (davetcoleman) [inline moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h]: This can just be your name. Not clear to me why you are listing OSRF since really GSoC is funding this project and Bryce/Felix are mentoring it.

File: moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+ *  modification, are permitted provided that the following conditions
+ *  are met:
+ *
+ *   * Redistributions of source code must retain the above copyright
+ *     notice, this list of conditions and the following disclaimer.
+ *   * Redistributions in binary form must reproduce the above
+ *     copyright notice, this list of conditions and the following
+ *     disclaimer in the documentation and/or other materials provided
+ *     with the distribution.
+ *   * Neither the name of Willow Garage nor the names of its
```
Comment (davetcoleman) [inline moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h]: Note the hidden Willow Garage here

File: moveit_core/collision_detection_bullet/src/collision_common.cpp [hunk 4]
```diff
@@ -0,0 +1,42 @@
...
+ *  POSSIBILITY OF SUCH DAMAGE.
+ *********************************************************************/
+
+/* Author: Jens Petit */
+
+#include <moveit/collision_detection_bullet/collision_common.h>
+#include <geometric_shapes/shapes.h>
+
+namespace collision_detection
+{
```
Comment (davetcoleman) [inline moveit_core/collision_detection_bullet/src/collision_common.cpp]: Add TODO explaining why this is empty

File: moveit_core/collision_detector_fcl_description.xml [hunk 5]
```diff
@@ -5,4 +5,4 @@
       FCL Collision Detector, default for MoveIt.
     </description>
   </class>
-</library>
```
Comment (davetcoleman) [inline moveit_core/collision_detector_fcl_description.xml]: ideally you revert this change, as it unnecessarily dirties the git history of this FCL file.

File: moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h (hunk 2 above)
Comment (j-petit) [reply moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h]: I thought it had to be a company

File: moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h (hunk 3 above)
Comment (j-petit) [reply moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h]: Thanks for reviewing so carefully Dave :+1:

**Comment by j-petit** [comment]
Close and open to restart Travis

**Comment by j-petit** [comment]
Can someone help me why clang-tidy fails?

**Comment by rhaschke** [comment]
@j-petit, the Travis failure is due to an incompatible merge of two PRs in https://github.com/ros-planning/moveit_ci. I will have a look...

**Comment by j-petit** [comment]
It passed after restarting Travis.

**Comment by j-petit** [comment]
Ping, can we get this merged?

**Comment by BryceStevenWilley** [comment]
@davetcoleman are you good with this PR?

File: moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h [hunk 6]
```diff
@@ -0,0 +1,55 @@
...
+#include <moveit/collision_detection_bullet/collision_world_bt.h>
+
+namespace collision_detection
+{
+/** \brief An allocator for Bullet collision detectors */
+class CollisionDetectorAllocatorBt
+    : public CollisionDetectorAllocatorTemplate<CollisionWorldBt, CollisionRobotBt, CollisionDetectorAllocatorBt>
+{
+public:
+  static const std::string NAME;  // defined in collision_world_bt.cpp
```
Comment (davetcoleman) [inline moveit_core/collision_detection_bullet/include/moveit/collision_detection_bullet/collision_detector_allocator_bt.h]:
```suggestion
  static const std::string COLLISION_CHECKER_NAME;  // defined in collision_world_bt.cpp
```

**Comment by j-petit** [comment]
Thanks!
